hangman.rb

```diff
@@ -15,17 +14,50 @@
 puts prompt
 puts "Your word has ", letter_array.length, "letters"
 puts prompt
-puts blank_array
+puts blank_array.join()
 puts prompt
 puts "You may begin guessing."
 puts prompt
-for x in 1..guesses
+
+  #gues1
+  iftrue = false
+  finish = true
+  game_end = false
+while guesses >= 0 && game_end == false
+  guess1 = STDIN.gets.chomp
+  iftrue = false
+  finish = true
+  game_end = false
   for i in 0..letter_array.length-1
-    if $stdin.gets.chomp == letter_array[i]
-      state = true
-    else
-      state = false
+
+    if guess1 == letter_array[i]
+      iftrue = true
```

Comment:
I see what you are doing here. The thing is you need to push the puts outside the `for loop`, otherwise it will print out multiple times if there are multiple correct letters.

`e l e p h a n t`

If we guess `e`, it will puts correct on index 0 and index 2. To avoid this we can move all the print logic outside the loop and handle it once the looping is done.